core/src/com/google/inject/internal/DefaultConstructionProxyFactory.java

```diff
@@ -46,16 +46,12 @@
 
     /*if[AOP]*/
     try {
-      net.sf.cglib.reflect.FastClass fc = BytecodeGen.newFastClassForMember(constructor);
-      if (fc != null) {
-        int index = fc.getIndex(constructor.getParameterTypes());
-        // We could just fall back to reflection in this case but I believe this should actually
-        // be impossible.
-        Preconditions.checkArgument(
-            index >= 0, "Could not find constructor %s in fast class", constructor);
-        return new FastClassProxy<T>(injectionPoint, constructor, fc, index);
+      BiFunction<Object, Object[], Object> fastConstructor =
+          BytecodeGen.fastConstructor(constructor);
+      if (fastConstructor != null) {
+        return new FastClassProxy<T>(injectionPoint, constructor, fastConstructor);
       }
-    } catch (net.sf.cglib.core.CodeGenerationException e) {
+    } catch (Exception | LinkageError e) {
```

Comment:
do we have any narrower type we can catch here, to avoid over-catching exceptions & suppressing real errors?

(same elsewhere we catch Exception or Throwable)